addons/mrp/models/mrp_production.py

```diff
@@ -1928,6 +1929,79 @@ def action_serial_mass_produce_wizard(self):
+                raise ValidationError(_("Only manufacturing orders in either the draft or confirmed status can be splitted."))
+            productions = [Command.create({'production_id': production.id}) for production in self]
+            wizard = self.env['mrp.production.split'].create({'production_ids': productions})
+            action = self.env['ir.actions.actions']._for_xml_id('mrp.action_mrp_production_split')
+            action['res_id'] = wizard.id
+            return action
+        else:
+            if self.state not in ('draft', 'confirmed'):
+                raise ValidationError(_("Only manufacturing orders in either the draft or confirmed status can be splitted."))
+            action = self.env['ir.actions.actions']._for_xml_id('mrp.action_mrp_production_split_production')
+            action['context'] = {
+                'default_production_id': self.id,
+            }
+            return action
+
+    def action_merge(self):
+        if len(self) == 1:
+            raise UserError(_("Nothing to Merge."))
+        if any(production.state not in ('draft', 'confirmed') for production in self):
+            raise ValidationError(_("Only manufacturing orders in either the draft or confirmed status can be merged."))
+        products = set([(production.product_id, production.bom_id) for production in self])
+        if len(products) > 1:
+            raise UserError(_('You can only merge manufacturing orders of identical products.'))
+        additional_raw_ids = self.mapped("move_raw_ids").filtered(lambda move: not move.bom_line_id)
+        additional_byproduct_ids = self.mapped('move_byproduct_ids').filtered(lambda move: not move.byproduct_id)
+        if additional_raw_ids or additional_byproduct_ids:
+            raise ValidationError(_("You can only merge manufacturing orders with no additional components or by-products."))
+        product_id, bom_id = products.pop()
+        users = set([production.user_id for production in self])
+        if len(users) == 1:
+            user_id = users.pop()
+        else:
+            user_id = self.env.user
+
+        origs = {}
+        for move in self.mapped('move_raw_ids'):
+            origs.setdefault(move.bom_line_id.id, []).extend(move.move_orig_ids.ids)
+        dests = {}
+        for move in self.mapped('move_finished_ids'):
+            dests.setdefault(move.byproduct_id.id, []).extend(move.move_dest_ids.ids)
+
+        production = self.env['mrp.production'].create({
+            'product_id': product_id.id,
+            'bom_id': bom_id.id,
+            'picking_type_id': bom_id.picking_type_id or self._get_default_picking_type(),
+            'product_qty': sum(production.product_uom_qty for production in self),
+            'product_uom_id': product_id.uom_id.id,
+            'user_id': user_id.id,
+            'origin': ",".join(sorted([production.name for production in self])),
+        })
+        self.env['stock.move'].create(production._get_moves_raw_values())
```

Comment:
@ticodoo It should be 2 since we recreate the move from the BoM. Anyway I already discussed it with mgm and I will prevent the flow of merge MO that has been manually edited. To avoid "undoing stuff" magically

Reply:
But I agree with you on the principle and merge MO manually edited is maybe for another task